testsuite/tests/ui/conftest.py

```diff
@@ -143,6 +180,50 @@ def navigator(browser):
     return navigator
 
 
+def generate_or_return(request, name):
+    """
+        Generates the new string, if the provided one is empty or None.
+    """
+    if name is None or name == "":
+        return blame(request, "name")
+    return name
+
+
+# pylint: disable=unused-argument, too-many-arguments
+@pytest.fixture(scope="module")
+def custom_ui_tenant(custom_master_login, navigator, threescale, testconfig, request, master_threescale):
+    """Parametrized custom Tenant created via UI"""
+
+    def _custom_ui_tenant(username: str = "",
+                          email: str = "",
+                          password: str = "",
+                          organisation: str = "",
+                          autoclean=True):
+
+        custom_master_login()
+        tenant = navigator.navigate(TenantNewView)
+
+        tenant.create(username=username, email=email+"@anything.invalid", password=password, organization=organisation)
+
+        account = master_threescale.accounts.read_by_name(organisation)
+        tenant = master_threescale.tenants.read(account.entity_id)
+        tenant.wait_tenant_ready()
+
+        if autoclean and not testconfig["skip_cleanup"]:
+            request.addfinalizer(tenant.delete)
+
+        return tenant
+
+    return _custom_ui_tenant
+
+
+@pytest.fixture(scope="module")
+def ui_tenant(custom_ui_tenant, request):
+    """Preconfigured tenant existing over whole testing session"""
+    name = blame(request, "ui_tenant")
+    return custom_ui_tenant(name, name)
```

Comment:
I think, this will not work

Reply:
U were right, it did not work for 2 reasons:

1. you cant have `_` in the forms
2. I was not providing the 4th (the most important) argument :D 

FIxed now + tested